<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Bioinformatics.jl Documentation · Bioinformatics Documentation</title><link href="https://cdnjs.cloudflare.com/ajax/libs/normalize/4.2.0/normalize.min.css" rel="stylesheet" type="text/css"/><link href="https://fonts.googleapis.com/css?family=Lato|Roboto+Mono" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.2.0/require.min.js" data-main="assets/documenter.js"></script><script src="siteinfo.js"></script><script src="../versions.js"></script><link href="assets/documenter.css" rel="stylesheet" type="text/css"/></head><body><nav class="toc"><h1>Bioinformatics Documentation</h1><select id="version-selector" onChange="window.location.href=this.value" style="visibility: hidden"></select><form class="search" id="search-form" action="search/"><input id="search-query" name="q" type="text" placeholder="Search docs"/></form><ul><li class="current"><a class="toctext" href>Bioinformatics.jl Documentation</a><ul class="internal"></ul></li></ul></nav><article id="docs"><header><nav><ul><li><a href>Bioinformatics.jl Documentation</a></li></ul><a class="edit-page" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/master/docs/src/index.md"><span class="fa"></span> Edit on GitHub</a></nav><hr/><div id="topbar"><span>Bioinformatics.jl Documentation</span><a class="fa fa-bars" href="#"></a></div></header><h1><a class="nav-anchor" id="Bioinformatics.jl-Documentation-1" href="#Bioinformatics.jl-Documentation-1">Bioinformatics.jl Documentation</a></h1><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.Sequence" href="#Bioinformatics.Sequence"><code>Bioinformatics.Sequence</code></a> — <span class="docstring-category">Type</span>.</div><div><div><p>Structure for sequence data.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/sequence.jl#L3">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.aliphatic_index-Tuple{Any,Any,Any,Float64}" href="#Bioinformatics.aliphatic_index-Tuple{Any,Any,Any,Float64}"><code>Bioinformatics.aliphatic_index</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function aliphatic_index(x_ala, x_val, x_ile, x_leu::Float64)</code></pre><p>The aliphatic index of a protein is defined as the relative volume occupied by aliphatic side chains (alanine, valine, isoleucine, and leucine).</p><p>See also: https://web.expasy.org/protparam/protparam-doc.html.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/stats.jl#L92-L99">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.dotmatrix-Tuple{Sequence,Sequence}" href="#Bioinformatics.dotmatrix-Tuple{Sequence,Sequence}"><code>Bioinformatics.dotmatrix</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function dotmatrix(s1::Sequence, s2::Sequence)</code></pre><p>Calculate the dotplot matrix for given two sequences.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/alignments.jl#L1-L5">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.edit_dist-Tuple{String,String}" href="#Bioinformatics.edit_dist-Tuple{String,String}"><code>Bioinformatics.edit_dist</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function edit_dist(s1::String, s2::String)</code></pre><p>Calculate edit distance between two strings.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/distances.jl#L1-L5">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.extinction_coeff-Tuple{Any,Any,Int64}" href="#Bioinformatics.extinction_coeff-Tuple{Any,Any,Int64}"><code>Bioinformatics.extinction_coeff</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function extinction_coeff(n_tyr, n_trp, n_cys::Int64)</code></pre><p>The extinction coefficient indicates how much light a protein absorbs at a certain wavelength. It is useful to have an estimation of this coefficient for following a protein which a spectrophotometer when purifying it.</p><p>See also: https://web.expasy.org/protparam/protparam-doc.html.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/stats.jl#L63-L71">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.frequency-Tuple{Sequence}" href="#Bioinformatics.frequency-Tuple{Sequence}"><code>Bioinformatics.frequency</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function frequency(seq::Sequence)</code></pre><p>Count bases / amino acids for given sequence.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/stats.jl#L1-L5">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.gc_content-Tuple{Sequence,Int64}" href="#Bioinformatics.gc_content-Tuple{Sequence,Int64}"><code>Bioinformatics.gc_content</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function gc_content(seq::Sequence, window_size::Int64)</code></pre><p>Calculate GC-content of given sequence using a window.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/stats.jl#L29-L33">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.gc_content-Tuple{Sequence}" href="#Bioinformatics.gc_content-Tuple{Sequence}"><code>Bioinformatics.gc_content</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function gc_content(seq::Sequence)</code></pre><p>Calculate GC-content of given sequence.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/stats.jl#L14-L18">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.gravy-Tuple{Sequence}" href="#Bioinformatics.gravy-Tuple{Sequence}"><code>Bioinformatics.gravy</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function gravy(seq::Sequence)</code></pre><p>The GRAVY value for a peptide or protein is calculated as the sum of hydropathy values of all the amino acids, divided by the number of residues in the sequence.</p><p>See also: https://web.expasy.org/protparam/protparam-doc.html.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/stats.jl#L104-L112">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.hamming_dist-Tuple{String,String}" href="#Bioinformatics.hamming_dist-Tuple{String,String}"><code>Bioinformatics.hamming_dist</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function hamming_dist(s1::String, s2::String)</code></pre><p>Calculate Hamming distance between two strings.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/distances.jl#L29-L33">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.instability_index-Tuple{Sequence}" href="#Bioinformatics.instability_index-Tuple{Sequence}"><code>Bioinformatics.instability_index</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function instability_index(seq::Sequence)</code></pre><p>The instability index provides an estimate of the stability of your protein in a test tube.</p><p>See also: https://web.expasy.org/protparam/protparam-doc.html.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/stats.jl#L76-L82">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.plot_dotmatrix-Tuple{Array{Int8,N} where N}" href="#Bioinformatics.plot_dotmatrix-Tuple{Array{Int8,N} where N}"><code>Bioinformatics.plot_dotmatrix</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function plot_dotmatrix(dotmatrix::Array{Int8})</code></pre><p>Plot dotplot for given dotmatrix.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/plots.jl#L14-L18">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.plot_gc_content-Tuple{Sequence,Int64}" href="#Bioinformatics.plot_gc_content-Tuple{Sequence,Int64}"><code>Bioinformatics.plot_gc_content</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function plot_gc_content(seq::Sequence, window_size::Int64)</code></pre><p>Plot GC-content of given sequence.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/plots.jl#L3-L7">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.possible_proteins-Tuple{Sequence}" href="#Bioinformatics.possible_proteins-Tuple{Sequence}"><code>Bioinformatics.possible_proteins</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function possible_proteins(s::Sequence)</code></pre><p>Find possible proteins of given sequence.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/sequence.jl#L113-L117">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.protein_mass" href="#Bioinformatics.protein_mass"><code>Bioinformatics.protein_mass</code></a> — <span class="docstring-category">Function</span>.</div><div><div><pre><code class="language-julia">function protein_mass(seq::Sequence, type = &quot;monoisotopic&quot;)</code></pre><p>Calculate mass of given amino acid sequence.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/stats.jl#L43-L47">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.protparam-Tuple{Sequence}" href="#Bioinformatics.protparam-Tuple{Sequence}"><code>Bioinformatics.protparam</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function protparam(seq::Sequence)</code></pre><p>Computes various physico-chemical properties that can be deduced from a protein sequence.</p><p>See also: https://web.expasy.org/protparam/protparam-doc.html.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/stats.jl#L197-L204">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.readFASTA-Tuple{String}" href="#Bioinformatics.readFASTA-Tuple{String}"><code>Bioinformatics.readFASTA</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function readFASTA(filename::String)</code></pre><p>Parse a FASTA formatted file.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/io.jl#L1-L5">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.reading_frames-Tuple{Sequence}" href="#Bioinformatics.reading_frames-Tuple{Sequence}"><code>Bioinformatics.reading_frames</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function reading_frames(s::Sequence)</code></pre><p>Find all reading frames of given sequence.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/sequence.jl#L95-L99">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.reverse_complement-Tuple{Sequence}" href="#Bioinformatics.reverse_complement-Tuple{Sequence}"><code>Bioinformatics.reverse_complement</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function reverse_complement(s::Sequence)</code></pre><p>Reverse complement of given DNA/RNA sequence.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/sequence.jl#L32-L36">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.transcription-Tuple{Sequence}" href="#Bioinformatics.transcription-Tuple{Sequence}"><code>Bioinformatics.transcription</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function transcription(dna_seq::Sequence)</code></pre><p>Transcribe DNA to RNA.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/sequence.jl#L20-L24">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.translation" href="#Bioinformatics.translation"><code>Bioinformatics.translation</code></a> — <span class="docstring-category">Function</span>.</div><div><div><pre><code class="language-julia">function translation(s::Sequence, start_pos::Int64 = 1)</code></pre><p>Translate given DNA sequence to Amino Acid sequence.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/sequence.jl#L77-L81">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.isoelectric_point-Tuple{Sequence}" href="#Bioinformatics.isoelectric_point-Tuple{Sequence}"><code>Bioinformatics.isoelectric_point</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">function isoelectric_point(seq::Sequence)</code></pre><p>The isoelectric point, is the pH at which a molecule carries no net electrical charge or is electrically neutral in the statistical mean.</p><p>See also: https://web.expasy.org/protparam/protparam-doc.html, http://isoelectric.org/algorithms.html.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/stats.jl#L120-L128">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.kmers" href="#Bioinformatics.kmers"><code>Bioinformatics.kmers</code></a> — <span class="docstring-category">Function</span>.</div><div><div><pre><code class="language-julia">function kmers(s::Sequence, k::Int64 = 2)</code></pre><p>Find all k-mers of given sequence.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/sequence.jl#L50-L54">source</a></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Bioinformatics.kmers_frequency" href="#Bioinformatics.kmers_frequency"><code>Bioinformatics.kmers_frequency</code></a> — <span class="docstring-category">Function</span>.</div><div><div><pre><code class="language-julia">function kmers_frequency(s::Sequence, k::Int64 = 2)</code></pre><p>Calculate frequencies of k-mers of given sequence.</p></div></div><a class="source-link" target="_blank" href="https://github.com/mrtkp9993/Bioinformatics.jl/blob/6c7b69397066a007c8d169f7d4d4540a3e9b4336/src/sequence.jl#L65-L69">source</a></section><footer><hr/></footer></article></body></html>
