var documenterSearchIndex = {"docs":
[{"location":"#Bioinformatics.jl-Documentation-1","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.jl Documentation","text":"","category":"section"},{"location":"#","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.jl Documentation","text":"CurrentModule = Bioinformatics\nDocTestSetup = quote\n    using Bioinformatics\nend","category":"page"},{"location":"#","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.jl Documentation","text":"Modules = [Bioinformatics]","category":"page"},{"location":"#Bioinformatics.Sequence","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.Sequence","text":"Structure for sequence data.\n\n\n\n\n\n","category":"type"},{"location":"#Bioinformatics.dotmatrix-Tuple{Sequence,Sequence}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.dotmatrix","text":"function dotmatrix(s1::Sequence, s2::Sequence)\n\nCalculate the dotplot matrix for given two sequences.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.edit_dist-Tuple{String,String}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.edit_dist","text":"function edit_dist(s1::String, s2::String)\n\nCalculate edit distance between two strings.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.frequency-Tuple{Sequence}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.frequency","text":"function frequency(seq::Sequence)\n\nCount bases / amino acids for given sequence.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.gc_content-Tuple{Sequence,Int64}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.gc_content","text":"function gc_content(seq::Sequence, window_size::Int64)\n\nCalculate GC-content of given sequence using a window.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.gc_content-Tuple{Sequence}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.gc_content","text":"function gc_content(seq::Sequence)\n\nCalculate GC-content of given sequence.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.gravy-Tuple{Sequence}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.gravy","text":"function gravy(seq::Sequence)\n\nThe GRAVY value for a peptide or protein is calculated as the sum of hydropathy values of all the amino acids, divided by the number of residues in the sequence.\n\nSee also: https://web.expasy.org/protparam/protparam-doc.html.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.hamming_dist-Tuple{String,String}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.hamming_dist","text":"function hamming_dist(s1::String, s2::String)\n\nCalculate Hamming distance between two strings.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.instability_index-Tuple{Sequence}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.instability_index","text":"function instability_index(seq::Sequence)\n\nThe instability index provides an estimate of the stability of your protein in a test tube.\n\nSee also: https://web.expasy.org/protparam/protparam-doc.html.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.plot_dotmatrix-Tuple{Array{Int8,N} where N}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.plot_dotmatrix","text":"function plot_dotmatrix(dotmatrix::Array{Int8})\n\nPlot dotplot for given dotmatrix.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.plot_gc_content-Tuple{Sequence,Int64}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.plot_gc_content","text":"function plot_gc_content(seq::Sequence, window_size::Int64)\n\nPlot GC-content of given sequence.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.possible_proteins-Tuple{Sequence}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.possible_proteins","text":"function possible_proteins(s::Sequence)\n\nFind possible proteins of given sequence.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.protein_mass","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.protein_mass","text":"function protein_mass(seq::Sequence, type = \"monoisotopic\")\n\nCalculate mass of given amino acid sequence.\n\n\n\n\n\n","category":"function"},{"location":"#Bioinformatics.protparam-Tuple{Sequence}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.protparam","text":"function protparam(seq::Sequence)\n\nComputes various physico-chemical properties that can be deduced from a protein sequence.\n\nSee also: https://web.expasy.org/protparam/protparam-doc.html.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.readFASTA-Tuple{String}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.readFASTA","text":"function readFASTA(filename::String)\n\nParse a FASTA formatted file.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.reading_frames-Tuple{Sequence}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.reading_frames","text":"function reading_frames(s::Sequence)\n\nFind all reading frames of given sequence.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.reverse_complement-Tuple{Sequence}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.reverse_complement","text":"function reverse_complement(s::Sequence)\n\nReverse complement of given DNA/RNA sequence.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.skew-Tuple{Sequence}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.skew","text":"function skew(seq::Sequence)\n\nCalculate the skew values of sequence.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.skew_plot-Tuple{Sequence}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.skew_plot","text":"function skew_plot(seq::Sequence)\n\nPlot skew diagram for given sequence.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.transcription-Tuple{Sequence}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.transcription","text":"function transcription(dna_seq::Sequence)\n\nTranscribe DNA to RNA.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.translation","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.translation","text":"function translation(s::Sequence, start_pos::Int64 = 1)\n\nTranslate given DNA sequence to Amino Acid sequence.\n\n\n\n\n\n","category":"function"},{"location":"#Bioinformatics.aliphatic_index-Tuple{Any,Any,Any,Float64}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.aliphatic_index","text":"function aliphatic_index(x_ala, x_val, x_ile, x_leu::Float64)\n\nThe aliphatic index of a protein is defined as the relative volume occupied by aliphatic side chains (alanine, valine, isoleucine, and leucine).\n\nSee also: https://web.expasy.org/protparam/protparam-doc.html.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.extinction_coeff-Tuple{Any,Any,Int64}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.extinction_coeff","text":"function extinction_coeff(n_tyr, n_trp, n_cys::Int64)\n\nThe extinction coefficient indicates how much light a protein absorbs at a certain wavelength. It is useful to have an estimation of this coefficient for following a protein which a spectrophotometer when purifying it.\n\nSee also: https://web.expasy.org/protparam/protparam-doc.html.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.isoelectric_point-Tuple{Sequence}","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.isoelectric_point","text":"function isoelectric_point(seq::Sequence)\n\nThe isoelectric point, is the pH at which a molecule carries no net electrical charge or is electrically neutral in the statistical mean.\n\nSee also: https://web.expasy.org/protparam/protparam-doc.html, http://isoelectric.org/algorithms.html.\n\n\n\n\n\n","category":"method"},{"location":"#Bioinformatics.kmers","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.kmers","text":"function kmers(s::Sequence, k::Int64 = 2)\n\nFind all k-mers of given sequence.\n\n\n\n\n\n","category":"function"},{"location":"#Bioinformatics.kmers_frequency","page":"Bioinformatics.jl Documentation","title":"Bioinformatics.kmers_frequency","text":"function kmers_frequency(s::Sequence, k::Int64 = 2)\n\nCalculate frequencies of k-mers of given sequence.\n\n\n\n\n\n","category":"function"}]
}
